openedx/core/djangoapps/credit/api.py

```diff
@@ -488,6 +487,70 @@ def is_credit_course(course_key):
     return CreditCourse.is_credit_course(course_key)
 
 
+def get_credit_requirement(course_key, namespace, name):
+    """Returns the requirement of a given course, namespace and name.
+
+    Args:
+        course_key(CourseKey): The identifier for course
+        namespace(str): Namespace of requirement
+        name(str): Name of the requirement
+
+    Returns: dict
+
+    Example:
+    >>> get_credit_requirement_status(
+        "course-v1-edX-DemoX-1T2015", "proctored_exam", "i4x://edX/DemoX/proctoring-block/final_uuid"
+        )
+            {
+                "course_key": "course-v1-edX-DemoX-1T2015"
+                "namespace": "reverification",
+                "name": "i4x://edX/DemoX/edx-reverification-block/assessment_uuid",
+                "display_name": "reverification"
+                "criteria": {},
+            }
+
+    """
+    requirement = CreditRequirement.get_course_requirement(course_key, namespace, name)
+    return {
+        "course_key": requirement.course.course_key,
+        "namespace": requirement.namespace,
+        "name": requirement.name,
+        "display_name": requirement.display_name,
+        "criteria": requirement.criteria
+    } if requirement else None
```

Comment:
Aside: I don't think this needs to be done now, but this kind of serialization would be best performed using DRF's [serializers](http://www.django-rest-framework.org/api-guide/serializers/).